src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -802,6 +804,12 @@ private async Task TryNotifyClientErrorAsync(IClientProxy client, string error,
         });
     }
 
+    private static void RestoreAntiforgeryToken(AsyncServiceScope scope)
+    {
+        var antiforgery = scope.ServiceProvider.GetService<AntiforgeryStateProvider>();
+        _ = antiforgery?.GetAntiforgeryToken();
+    }
+
```

Comment:
Trying to retrieve the token during the initial render so that we capture it.

Reply:
I don't think I understand why `GetAntiforgeryToken()` needs to be invoked. Does it have some side effect we rely on? If it's an `EndpointAntiforgeryStateProvider`, are we calling `GetAntiforgeryToken()` because it calls `IAntiforgery.GetAndStoreTokens(_context)`?

Would it be possible to directly invoke the missing functionality here rather than relying on `GetAntiforgeryToken()` having a specific implementation? If not, I think it might be worth adding a comment that explains what's going on and why it's correct.

Reply:
Calling `GetAntiforgeryToken` forces the token to be populated. I think it's best if we just call `GetAntiforgeryToken` rather than trying to do something more sophisticated.

> GetAntiforgeryToken() because it calls IAntiforgery.GetAndStoreTokens(_context)?

It does not, in interactive cases it restores the token from persistent component state, but it does so at the time GetAntiforgeryToken is invoked. Since this component is responsible for storing it, it should be responsible for restoring it.

Reply:
I've added a comment